Evolutionary_loop/Archived/ssPermissionCheck.py

```diff
@@ -0,0 +1,53 @@
+#*******************************************************************************
+#  file: ssPermissionCheck.py
+#     This script is based on ssPermissionCheck.sh
+#     This is an attempt to slowly migrate to Python.
+#
+#  Programmer: Jason Yao (yao.966@osu.edu)
+#
+#  Revision history:
+#     07/18/23  Original version
+#
+#  Notes:
+#     * vertical ruler at column 80
+#
+#  TODO: 
+#     * ssPermissionCheck.sh seems capable of waiting (sleep 1m);
+#       add this in the future.
+#     * ssPermissionCheck.sh sends email with error messages; add this.
+#
+#*******************************************************************************
+import argparse as ap
+
+### ARGUMENTS ###
+# Use the following command to checkout the arguments in the terminal
+# >> Python3 ssPermissionCheck.py -h
+parser = ap.ArgumentParser()
+parser.add_argument("RunName", help="Run Name")
+parser.add_argument("WorkingDir", help="Working Directory (Evolutionary_Loop)")
+parser.add_argument("usr_spec", help="User-specified checkpoint (aka state)")
```

Comment:
`type` added.
Not sure how `default` can be implemented for positional arguments.

Reply:
You should be able to do it by adding `nargs='?'` to make the positional args optional.  However since they are positional, you can't specify a specific arg, but only:
* no args: uses all 3 default values
* 1 arg (`run_name`): uses the default value for `working_dir` and `usr_spec`
* 2 args (`run_name` and `working_dir`): uses the default value for `usr_spec`
* 3 args: uses all the 3 args passed by the command line and no default

IOW there's no way to specifically pass `usr_spec` and use the default value for the first two args.

Reply:
Got it; will keep this in mind for future reference.